addons/stock/models/stock_move.py

```diff
@@ -1039,7 +1039,6 @@ def _push_apply(self):
             if move.location_dest_id.company_id != self.env.company:
                 ProcurementGroup = self.env['procurement.group'].sudo()
                 move = move.with_context(allowed_companies=self.env.user.company_ids.ids)
-                warehouse_id = False
```

Comment:
Might be nice to at least describe the issue a little bit in the commit message. :smile: 

Maybe it would be nice to add a test that uses a route that goes further than Customers to make sure it triggers properly as well, now that it checks the warehouse routes again.